sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2505,7 +2505,6 @@ object SQLConf {
 
   val LEGACY_REPLACE_DATABRICKS_SPARK_AVRO_ENABLED =
     buildConf("spark.sql.legacy.replaceDatabricksSparkAvro.enabled")
```

Comment:
@gengliangwang do we need this conf? We do the same thing in `com.databricks.spark.csv` already by default internally. We could just make it exposed and deprecate this config.

Reply:
In this way, I think it will address most of concerns raised.

Reply:
And then we could remove both `com.databricks.spark.csv` and `com.databricks.spark.avro` fallbacks together in the future, of course (Spark 4.0?). I don't think it makes sense to keep both mapping forever.

Reply:
Let's deprecate it and then remove it in later versions. It's OK if a deprecated internal config is mentioned in public docs for migration purpose.

Reply:
Here is the PR https://github.com/apache/spark/pull/31578 to deprecate the config.

Reply:
Thank you all!